src/Helpers/Compress.php

```diff
@@ -0,0 +1,125 @@
+     * @return callable The middleware
+     */
+    private static function compressor(int $threshold, array $allowedEncodings, ?string $encoding): callable
+    {
+        return function (Context $ctx, callable $next) use ($threshold, $allowedEncodings, $encoding) {
+            $next($ctx);
+
+            $response = $ctx->getResponse();
+            $contentLength = $response->getHeaderLine('Content-Length');
+
+            if (
+                $response->hasHeader('Content-Encoding') ||
+                $ctx->req->method() === 'HEAD' ||
+                ($contentLength && (int) $contentLength < $threshold) ||
+                !self::shouldCompress($response) ||
+                !self::shouldTransform($response)
+            ) {
+                return $response;
+            }
+
+            $acceptedEncodings = explode(',', $ctx->req->header('Accept-Encoding'));
+            if (!$encoding) {
+                foreach ($allowedEncodings as $enc) {
+                    if (in_array($enc, $acceptedEncodings)) {
+                        $encoding = $enc;
+                        break;
+                    }
+                }
+            }
+
+            if (!$encoding || !$response->getBody()) {
+                return $response;
+            }
+
+            $compressedBody = self::performCompression($response->getBody(), $encoding);
+            $response = $response->withBody($compressedBody)
+                ->withoutHeader('Content-Length')
+                ->withHeader('Content-Encoding', $encoding);
+
+            return $response;
+        };
+    }
+
+    /**
+     * Check if the response should be compressed based on the Content-Type header
+     *
+     * @param ResponseInterface $response The response to check
+     * @return bool Whether the response should be compressed
+     */
+    private static function shouldCompress(ResponseInterface $response): bool
+    {
+        $type = $response->getHeaderLine('Content-Type');
+        return preg_match(self::STR_REGEX, $type);
+    }
+
+    /**
+     * Check if the response should be transformed (e.g. compressed) based on the
+     * Cache-Control header
+     *
+     * @param ResponseInterface $response The response to check
+     * @return bool Whether the response should be transformed
+     */
+    private static function shouldTransform(ResponseInterface $response): bool
+    {
+        $cacheControl = $response->getHeaderLine('Cache-Control');
+        return !preg_match('/(?:^|,)\s*?no-transform\s*?(?:,|$)/i', $cacheControl);
+    }
+
+    /**
+     * Perform the actual compression of the response body using the specified encoding
+     *
+     * @param string $body The response body to compress
+     * @param string $encoding The encoding to use for compression
+     *
+     * @return string|StreamInterface The compressed response body
+     */
+    private static function performCompression($body, string $encoding)
```

Comment:
Do we need a try/catch block here, a return type, and support for `br`?

For example:

```php
 try {
        $compressed = match ($encoding) {
            'gzip' => gzencode((string)$body),
            'deflate' => gzdeflate((string)$body),
            'br' => extension_loaded('brotli') ? brotli_compress((string)$body) : (string)$body,
            default => (string)$body,
        };
        return Utils::streamFor($compressed);
    } catch (\Exception $e) {
        // Log the error
        error_log("Compression failed: " . $e->getMessage());
        return Utils::streamFor((string)$body);
    }
```

Reply:
I never use `brotli` but it's look good. And also the try/catch, I think we don't need that cz the default fallback is string CMIIW.

Reply:
Sounds good!

In that case, we could do something like this:

```php
$content = (string) $body;
        return match ($encoding) {
            'br' => Utils::streamFor(self::brotliCompress($content)),
            'gzip' => Utils::streamFor(gzencode($content)),
            'deflate' => Utils::streamFor(gzdeflate($content)),
            default => $body,
        };
```

Which with the `brotliComrpess` method raising an error if it's not avalable:

```php
private static function brotliCompress(string $content): string|false
    {
        if (!function_exists('brotli_compress')) {
            throw new \RuntimeException('Brotli compression is not available. Make sure the Brotli PHP extension is installed and enabled.');
        }
        return brotli_compress($content);
    }
```

Reply:
Wooo... you got it bro!!!! :dancers: